main/tests/server/src/com/google/refine/importing/ImportingUtilitiesTests.java

```diff
@@ -110,6 +112,72 @@ public void testAllocateFileDeduplication() throws IOException {
         Assert.assertEquals(allocated, new File(dirA, "dummy-2"));
     }
 
+    @Test
+    public void testNormalizePath() {
+        // Test for issue Unable to create a project from a URL on Windows if the URL path contains ":" character #4625
+        // https://github.com/OpenRefine/OpenRefine/issues/4625
+        String urlPath = "/a/b:c/dummy:test";
+        String urlPathFixed = "\\a\\b-c\\dummy-test";
```

Comment:
I am surprised that a forward slash is converted to a backwards slash here - does that actually work? If we originally have a local path, then the normalization function should probably also return a local path, no?

Reply:
I have check this in the `File` class. And I found that when I create a new object by a path (in OpenRefine the path is from the method `url.getpath()`). The bug I fixed is actually here, using this path to create a file. In the construct, it will check the current Operation System and create a corresponding file with the path passed in if the path doesn't contain the illegal charater in current system. If not, there will be an crash. That's the issue about. And this work well in linux and mac but not well in windows. To make minimum impact on OpenRefine, my modification only work when current system is Windows, otherwise it will return to the original path. 
- For the first question, I have tested this and it work well. And the convertion only occurs in Windows. To implement the convertion, I use the library `org.apache.commons.io.FilenameUtils`  and the eaxct method ` String pathWithWSeparator = FilenameUtils.separatorsToWindows(path)`.
 - For the second question, I think it doesn't matter because the method `static public File allocateFile(File dir, String name)`  invokes this normalize. And the `allocateFile()` method only return a file object, it doesn't affect the method. Since if normalize was invoked means the current System is Windows, and the path in the file object is  separated by backward slash.  It works. If the current system is not Windows, it won't invoke the normalize method. It also works.